content/docs/architecture/auth.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Authentication and Authorization
+weight: 60
+---
+
+
+Fluvio uses mTLS (mutual TLS) for authentication between client and the fluvio cluster.   The user id (or principle) is extracted from the client certificate then apply to authorization policy.  Currently, Fluvio supports a simple RBAC (role based access control) authorization model but it could be extended to support more complex authorization scheme in the future. 
+
+# Cluster setup TLS
+
+First Fluvio must be running with TLS enabled. Once TLS is enabled, Both SC and SPU will accept mTLS base traffic from the client.  As part of TLS based configuration, admin can also specify authorization policy file.
+
+To create a Fluvio cluster with TLS enabled, you must provide following certs and keys:
+- CA cert
+- Client cert and key
+- Server cert and key
+- Authorization policy file
+
+For development and testing, you can use self-signed certs and keys.  For production, you should use certs and keys signed by a trusted CA.  It is important to note that the client cert and key must be signed by the same CA as the server cert and key.  Otherwise, the client will not be able to connect to the server.  In addition, admin must specify domain (DNS name) that cluster will be bind to.  This is required for TLS to work properly.
+
+## Self signed certs
+
+For self-signed certs, you can use sample certs templates in [fluvio](https://github.com/infinyon/fluvio/tree/master/tls).  
+
+## Authorization policy file
+
+Fluvio current implements a simple type level permission model.  The authorization policy file is a JSON file that contains a list of permission entries for the role. Each entry contains a type and a list of permissions.  Tye type can be one of the following object types.  Current list of types are:
+-  CustomSpu
+-  Spu
+-  Topic
+-  Partition
+-  SpuGroup
+-  SmartModule
+-  TableFormat
+
+Each object type can have one or more permissions.
+- All
+- Read
+- Write
+
+The `All` permission means the user has all permissions for the object type.   For example, following is a policy file.  It contains a two roles: `Root` and `User`.  The `Root` role has all permissions for all object types.  The `TopicUser` role only has read permission for `Topic` object type.
+
+```json
+{
+    "Root": {
+        "CustomSpu": [
+            "All"
+        ],
+        "Partition": [
+            "All"
+        ],
+        "Spu": [
+            "All"
+        ],
+        "Topic": [
+            "All"
+        ],
+        "SpuGroup": [
+            "All"
+        ],
+        "ManagedConnector": [
+            "All"
+        ],
+        "SmartModule": [
+            "All"
+        ],
+        "TableFormat": [
+            "All"
+        ]
+    },
+    "TopicUser": {
+        "Topic": [
+            "Read"
+        ]
+    }
+
+}
+```
+
+Note that, in order to admin the cluster, "Root" role with full access to all objects is required.
```

Comment:
<s>in order to admin the cluster</s> to avoid locking yourself out of the system, the "Root" keyword must have "All" access to all system objects.

Reply:
updated